python/hail/context.py

```diff
@@ -147,6 +149,32 @@ def version(self):
         """
         return self._jhc.version()
 
+    @property
+    def default_reference(self):
+        """Return the default reference genome.
+
+        The default reference genome is set to :py:meth:`hail.GenomeReference.GRCh37`,
+        but can be changed to a different reference with :py:meth:`~hail.HailContext.set_default_reference`.
+
+        :rtype: :class:`.GenomeReference`
+        """
+
+        if not self._default_ref:
+            self._default_ref = GenomeReference._from_java(Env.hail().variant.GenomeReference.defaultReference())
+        return self._default_ref
+
+    @handle_py4j
+    @typecheck_method(reference_genome=GenomeReference)
+    def set_default_reference(self, reference_genome):
```

Comment:
Hmm.  I think the default should be on the HailContext constructor.  It seems dangerous to modify.  I was thinking users would use the default reference is they were using only one reference (or primarily one reference), and use the individual function overrides if they were mixing references or needed the occasional exception.  In either case, I didn't imagine they could change the default reference.

Also, all things being equal, it is better to make things immutable.

Feel free to push back.

Reply:
The issue I was concerned about is say I want to create my own reference `foo`. Then I have a chicken and egg problem where the GenomeReference requires a running HailContext to initiate the Java object, but the HailContext is expecting a GenomeReference as an argument. 

I agree having the reference as a parameter is safer. Is there a way to do this?

Reply:
Good point.  Hmm.

I'm wondering if anyone will ever create a reference grammatically in Python or they'll always load it from a file.  Where does `addReference` get called now?  I can't find any users of it.  It looks like you can load from a resources but you don't let the  user load from a file.  What if you let the user specify a reference either from a JSON file or the equivalent Python data, and let the HailContext take that?  Then GenomeReference should have an option to construct from the same data in the general case (and maybe get rid of the constructor that takes the toplevel fields as separate arguments.)

What do you think?